modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/Applicative.kt

```diff
@@ -18,4 +18,13 @@ interface Applicative<F> : Apply<F> {
 
   override fun <A, B> Kind<F, A>.map(f: (A) -> B): Kind<F, B> =
     ap(just(f))
+
+  fun <A> Kind<F, A>.replicate(n: Int): Kind<F, List<A>> =
+    if (n <= 0) just(emptyList())
+    else map(this, replicate(n - 1)) { (a, xs) -> listOf(a) + xs }
+
+  fun <A> Kind<F, A>.replicate(n: Int, MA: Monoid<A>): Kind<F, A> = MA.run {
+    if (n <= 0) just(empty())
+    else map(this@replicate, replicate(n - 1, MA)) { (a, xs) -> a + xs }
```

Comment:
Small nit, use `MA.run` just locally in the block inside map. Use `MA.just`. This is because the Android compiler has problem with nested `this` scope, so we play it safe.

Reply:
Done

Reply:
Merge at will! You have rights, IIRC :D

Reply:
Will do ^^ Thanks